test/ssl_test.c

```diff
@@ -160,25 +160,25 @@ static int check_compression(HANDSHAKE_RESULT *result, SSL_TEST_CTX *test_ctx)
 static int check_npn(HANDSHAKE_RESULT *result, SSL_TEST_CTX *test_ctx)
 {
     int ret = 1;
-    ret &= strings_equal("NPN Negotiated (client vs server)",
-                         result->client_npn_negotiated,
-                         result->server_npn_negotiated);
-    ret &= strings_equal("ExpectedNPNProtocol",
-                         test_ctx->expected_npn_protocol,
-                         result->client_npn_negotiated);
+    ret &= strings_eq(result->client_npn_negotiated,
```

Comment:
Collecting the errors wouldn't be too hard.  A stack of strings, malloc buffers and sprintf into them.
Fail if the stack isn't empty.